File: tslang/test/array/insert-middle/ArrayInsertMiddle.test.ts
import { expect, test } from "bun:test";
import { arrayInsertMiddle } from "../../../src/array/insert-middle/ArrayInsertMiddle";

test("test ArrayInsertMiddle", () => {
	expect(arrayInsertMiddle([1, 2, 3, 9, 5], 8)).toEqual([1, 2, 3, 8, 9, 5]);
	expect(arrayInsertMiddle([1, 2, 3, 4], 9)).toEqual([1, 2, 9, 3, 4]);
});

File: ktlang/app/bin/test/ktlang/array/insert-middle/ArrayInsertMiddleTest.kt
import kotlin.test.Test
import kotlin.test.assertEquals
import ktlang.array.insert.middle.*

class ArrayInsertMiddleTest {

    @Test
    fun testAcertMiddle() {
        val result = arrayInsertMiddle(listOf(1, 2, 3, 9, 5), 8)
        assertEquals(listOf(1, 2, 3, 8, 9, 5), result)
        // assertEquals(listOf(1, 2, 9, 3, 4), arrayInsertMiddle(listOf(1, 2, 3, 4), 9))
    }
}
